minimesos/src/main/java/com/containersol/minimesos/cluster/MesosCluster.java

```diff
@@ -168,9 +168,28 @@ public void install(String marathonJson) {
         String marathonIp = marathon.getIpAddress();
         LOGGER.debug(String.format("Installing %s app on marathon %s", marathonJson, marathonIp));
 
-        marathon.deployApp(marathonJson);
+        marathon.deployApp(replaceTokens(marathonJson));
     }
 
+    public String replaceTokens(String marathonJson) {
+        // received JSON might contain tokens, which should be replaced before the installation
+        List<ClusterProcess> uniqueRoles = ClusterUtil.getDistinctRoleProcesses(memberPocesses);
+        for( ClusterProcess process : uniqueRoles ) {
+            URI serviceUri = process.getServiceUrl();
+            if (serviceUri != null) {
+                marathonJson = replaceToken(marathonJson, "MINIMESOS_" + process.getRole().toUpperCase(), serviceUri.toString());
+                marathonJson = replaceToken(marathonJson, "MINIMESOS_" + process.getRole().toUpperCase() + "_IP", serviceUri.getHost());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "marathonJson". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS1226)